Framework/Geometry/src/Instrument/ComponentInfoBankHelpers.cpp

```diff
@@ -0,0 +1,27 @@
+#include "MantidGeometry/Instrument/ComponentInfoBankHelpers.h"
+#include "MantidBeamline/ComponentType.h"
+#include "MantidGeometry/Instrument/ComponentInfo.h"
+
+using Mantid::Beamline::ComponentType;
+
+namespace Mantid {
+namespace Geometry {
+namespace ComponentInfoBankHelpers {
+bool isDetectorFixedInBank(const ComponentInfo &compInfo,
+                           const size_t detIndex) {
+  auto parent = compInfo.parent(detIndex);
+  auto grandParent = compInfo.parent(parent);
+  auto grandParentType = compInfo.componentType(grandParent);
+
```

Comment:
You could also test that `parent` and `grandParent` are unique indexes and are distinct from `detIndex`. If your detector was attached directly to the instrument, there would be no grandparent, and the parent and grandparent indexes would be the same. So could just return false at that point. Maybe this isn't worth doing though?

Reply:
The test would fail as long as the grandparent type isn't match regardless.